src/PokemonDetail.jsx

```diff
@@ -0,0 +1,199 @@
+import { Link, useParams } from "react-router-dom";
+import { useFetch } from "./hooks/useFetch";
+import { useContext, useState } from "react";
+import { ThemeContext } from "./ThemeContext";
+import { titleCase } from "./functions/titleCase";
+import { getTypeString } from "./functions/getTypeString";
+import { Modal } from "./components/Modal";
+import store from "./store/store";
+
+export const PokemonDetail = () => {
+	const [showModal, setShowModal] = useState(false);
+
+	const handleShowModal = () => {
+		setShowModal(true);
+	};
+
+	const handleHideModal = () => {
+		setShowModal(false);
+	};
+
+	const buttonClickHandler = (newItem) => {
+		let ok = true;
```

Comment:
Same suggestion, I think it might be easier to disable the button. This logic around finding if it has been favorited could likely be shared by [calculating derived data with a selector](https://redux.js.org/usage/deriving-data-selectors#calculating-derived-data-with-selectors)